### ISSUE TITLE
Use of `@key` and `@ref` with `<persName>`

### ISSUE THREAD

**Issue by michaelnmmeyer** [erc-dharma/project-documentation#252]
For referencing person names in `<teiHeader>`, we use the attribute `@ref`, as in:

```
<persName ref="part:jodo">
<forename>John</forename>
<surname>Doe</surname>
</persName>
```

However, the EGD also prescribes to use `@key`, under section "Tagging contemporary names":

```
<persName key="part:argr">
<forename>Arlo</forename>
<surname>Griffiths</surname>
</persName>
```

... and under section "Identifying persons and places with `@key`":

```
<persName key="Śiva">Pinākin</persName>
```

Is this intended?

For now, there is  a single occurrence of `<persName>` with `@key` in our corpus:

```
<persName key="hermit">Śivatejaḥ</persName>
```

... and there are 7414 occurrences of `<persName>` with `@ref`. They have the following 4 basic forms:

```
<persName ref="part:jodo">
<persName ref="http://viaf.org/viaf/66465311">
<persName ref="pallava000004">
<persName ref="PSN01005">
```

**Comment by danbalogh** [comment]
Good question. It seems to me that the one in EGD§10.5.1 is now an error, since it conflicts what is being done in the files. I know that originally, `@key` was used for this purpose. Checking old EGD comments, I find only mentions of `@key` from December 2019. I have no idea when it was changed to `@ref`, but either I forgot to change this in the EGD or I was never informed that the change has been made. I've now tagged this in the GDoc EGD and 

For the other use of `@key`, for historical `<persName>` (and `<placeName>` ) as per EGD §10.6.3 (like Śivatejaḥ), the last I know is that we agreed to use `@key` and then, at some point, harvest the values occurring in the text and consider developing them into a prosopography and changing to `@ref` to point to that. I have never used name encoding and I don't know if anybody's used this, or if they have decided to change over to key without telling me. I don't mind revising the relevant EGD section if the people who do encode names prefer it that way.

For the four basic forms of persName with ref, I'm only familiar with the first, which is the only one covered in the EGD. The second (with a viaf url) seems OK to me, but I don't know if any guide calls for such a reference; it is certainly not in the EGD. The fourth (pallava#####) seems to be an error meant to be a reference to a Pallava inscription, but I don't know. And I have no idea what the last is.

**Comment by michaelnmmeyer** [comment]
OK, thanks. For the historical use of `<persName>`, I think we can keep recommending `@key` instead of `@ref`, to  be coherent with the way `<placeName>` is used. There are not much occurrences of this kind, and they apparently all belong to Manu and Amandine, so I will straighten things out with them.

**Comment by danbalogh** [comment]
Thanks, I agree.

**Comment by arlogriffiths** [comment]
Dear @danbalogh — as for "The second (with a viaf url) seems OK to me, but I don't know if any guide calls for such a reference; it is certainly not in the EGD", I think I am to blame. I think I invented this and asked Kunthea and others to use it when encoding an edition that basiclaly reproduces the work of George Cœdès. Now I am about to encode an inscription edited in collaboration with Amandine Lepoutre, who has left scholarship and is not a member of DHARMA. I can either rcheat and create a DHARMA part: id for her, or encode her intellectual responsability with viaf url, or just claim responsability myself. We have discussed the issues many times before and as I recall you have made a concrete suggestion for how to revise EGD on this issue. Can you remind me where you have written up your thoughts? 

It would be very nice if a final decision could be taken soon, and EGD as well as template revised on the relevcant points.

**Comment by danbalogh** [comment]
Dear @arlogriffiths , my earlier suggestions (still with many details to clear up) about encoding responsibility are in the Leftovers doc under [Authorship and responsibility](https://docs.google.com/document/d/1ZXM8qL67DjPMScXUMV2g9syLyPraA2ZKmRzcybJ637Q/edit#heading=h.ay66e1w4gx8u). This is probably what you had in mind. But it's only about roles and responsibilities and has nothing about machine-actionable identifiers for non-Dharma members. I have never, as far as I recall, thought about the latter and have no opinion, preference or suggestion. As far as I am concerned, EGD §10.5.1 is sufficient:

- wrap name elements in `<forename>` and `<surname>` or, if these are not applicable, the entire name in `<name>`
- wrap the above in `<persName>`
- if (and only if) it's a project member, add `@ref` with the ID

If you feel the need for more, I think we should keep it optional and loose. If @michaelnmmeyer says this is technically fine, then I would suggest `@ref` with any URI that identifies a person. If some resource lists are to be preferred, it's for our metadata experts to say which those should be. At a guess, it could be ORCID for currently active scholars, and wikidata for everyone else. I'm not familiar with VIAF, but it seems to be pretty messy to the non-initiated eye, with a bunch of listings for e.g. John Faithfull Fleet and many more for George Cœdès.

**Comment by danbalogh** [comment]
@arlogriffiths : roles and responsibilities have been sorted out and finalised, but there remains the question of identifying people who don't have a DHARMA ID. I stand by my above comment (Apr 8 2024) and prefer not to change the EGD but can be persuaded to add an option for an external reference (any URI) for identification.

**Comment by arlogriffiths** [comment]
I think our choices regarding the division of labor between `<respStmt>` and epigraphic lemma have removed the need for most such cases to appear, though I can still think of cases like the PYU files encoded by Marc Miyake that have been absorbed and in most cases till await my attention for manual adjustment to DHARMA norms. When I do, I will want to keep Marc Miyake in `<respStmt>` and can then do so following the method implied by your 3rd bullet, namely not adding and @ref because the person in question is not a project member.

In short, I concur with your comment of Apr 8 2024. You can close this thread if all the needful has been done in revising EGD.
